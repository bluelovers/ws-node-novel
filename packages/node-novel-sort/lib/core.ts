/**
 * Created by user on 2020/6/5.
 */

import naturalCompare from '@bluelovers/string-natural-compare';
import { ICreateSortCallbackOptions, IFnSortCallback, EnumToLowerCase } from './types';
import { _match, _trim } from './util';

/**
 * create a compare callback by (transpileBase value) -> trigger(transpile value) -> failbackSort
 * @param options
 */
export function createSortCallback(options: ICreateSortCallbackOptions = {}): IFnSortCallback
{
	const r = options.dotNum ? /^(\d+(?:\.\d+)?)/ : /^(\d+)/;

	const failbackSort = options.failbackSort || naturalCompare;
	const trigger = options.trigger || _match;
	let transpile = options.transpile || _trim;
	let transpileBase = options.transpileBase;

	if (options.toLowerCase)
	{
		let fnLowerCase: (input: string, ...argv) => string;

		if (typeof options.toLowerCase === 'function')
		{
			fnLowerCase = options.toLowerCase;
		}
		else
		{
			let fn = 'toLowerCase';

			if (typeof options.toLowerCase === 'number')
			{
				if (options.toLowerCase !== EnumToLowerCase.toLowerCase)
				{
					fn = 'toLocaleLowerCase';
				}
			}

			fnLowerCase = (input: string, ...argv) => input[fn](...argv);
		}

		if (fnLowerCase)
		{
			if (transpileBase)
			{
				transpileBase = ((old) =>
				{
					return function (input, ...argv)
					{
						return fnLowerCase(old(input, ...argv), ...argv)
					}
				})(transpileBase);
			}
			else
			{
				transpileBase = fnLowerCase;
			}
		}
	}

	let fnSortCallback: IFnSortCallback = function fnSortCallback(a: string, b: string, isSub?: boolean): number
	{
		if (a === b)
		{
			return 0;
		}

		let ret = trigger(transpile(a, isSub), transpile(b, isSub), {
			r,
			mainFn: fnSortCallback as IFnSortCallback,
			isSub,
		});

		return (typeof ret == 'number') ? ret : failbackSort(a, b);
	} as IFnSortCallback;

	if (transpileBase)
	{
		fnSortCallback = (function (oldFn)
		{
			return function (a: string, b: string, isSub?: boolean): number
			{
				if (a === b)
				{
					return 0;
				}

				if (isSub)
				{
					return oldFn(a, b, isSub);
				}

				return oldFn(transpileBase(a), transpileBase(b), isSub);
			} as IFnSortCallback
		})(fnSortCallback);
	}
	else
	{
		transpileBase = (input: string) => input;
	}

	fnSortCallback.failbackSort = failbackSort;
	fnSortCallback.trigger = trigger;
	fnSortCallback.transpile = transpile;
	fnSortCallback.transpileBase = transpileBase;
	fnSortCallback.fnSortCallback = fnSortCallback;

	return fnSortCallback;
}
