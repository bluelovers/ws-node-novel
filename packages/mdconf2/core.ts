/**
 * Module dependencies.
 */
import md, { Token, Tokens } from 'marked';
import { crlf, LF, CRLF, CR, chkcrlf } from 'crlf-normalize';
import { isMoment } from 'moment';
import isPlainObject from 'is-plain-object';
import { RawObject, IRawObjectTokenPlus, ITokenText2, IRawObjectPlus, IRawObjectDataPlus } from './lib/RawObject';
import { createInlineLexer, makeCodeBlock, normalize, put, getobjectbyid } from './lib/core';

export interface IOptionsParse
{
	/**
	 * @deprecated
	 */
	crlf?: typeof LF | typeof CRLF | typeof CR,

	oldParseApi?: boolean,

	allowBlockquote?: boolean,

	disableKeyToLowerCase?: boolean,

	markedOptions?: md.MarkedOptions,

	filterObjectKey?,
}

export const defaultOptionsParse: IOptionsParse = {
	crlf: LF,
	allowBlockquote: true,

	markedOptions: Object.assign({},
		md.defaults,
		{
			breaks: true,
		},
	),
};

export interface IObjectParse
{
	[key: string]: any
}

/**
 * Parse the given `str` of markdown.
 *
 * @param {String | Buffer} str
 * @param {Object} options
 * @return {Object}
 * @api public
 */
export function parse(str: string, options?: IOptionsParse): IObjectParse
export function parse(str: Buffer, options?: IOptionsParse): IObjectParse
export function parse(str: string | Buffer, options: IOptionsParse = {}): IObjectParse
{
	{
		let markedOptions = Object.assign({}, defaultOptionsParse.markedOptions, options.markedOptions);

		options = Object.assign({}, defaultOptionsParse, options, {
			markedOptions,
		});
	}

	let source: string = str.toString();
	let eol: typeof LF | typeof CRLF | typeof CR;

	if (1)
	{
		// disable crlf options
		eol = LF;
		source = crlf(source, eol);
	}
	/*
	else if (options.crlf)
	{
		eol = options.crlf;
		source = crlf(source, eol);
	}
	else
	{
		let ck = chkcrlf(source);
		eol = ck.lf ? LF : (ck.crlf ? CRLF : CR);
	}
	 */

	let lexer = new md.Lexer(options.markedOptions);

	let toks = lexer.lex(source);
	let conf = {};
	let keys: string[] = [];
	let depth = 0;
	let inlist = false;

	let paragraph: string[] = [];
	let paragraph2: string[] = [];
	let last_tok: md.Token;
	let blockquote_start: boolean;

	let inline_lexer = createInlineLexer(toks, Object.assign({}, options, {

	}));

	/*
	let _inline_md = new MarkdownIt({
		linkify: false,
	});
	*/

	(toks as Token[]).forEach(function (tok, index)
	{
		let val: string = (tok as Tokens.Code).text ;
		let _skip: boolean;
		let type: (Token | ITokenText2)["type"] = tok.type;

		if (type == 'text' && val.match(/[a-z]+\:\/\//i))
		{
			let r = inline_lexer.output(val);
			//let r = _inline_md.renderInline(val);

			if (val !== r && /^\s*<a href=/.test(r))
			{
				type = 'text2';
			}
		}

		switch (tok.type)
		{
			case 'heading':
				while (depth-- >= tok.depth) keys.pop();
				keys.push(normalize(tok.text, options));
				depth = tok.depth;

				paragraph = [];

				break;
			case 'list_item_start':
				inlist = true;
				break;
			case 'list_item_end':
				inlist = false;
				break;
				// @ts-ignore
			case 'text2':
			case 'text':
				put(conf, keys, tok.text, undefined, undefined, options, {
					type,
				});
				break;
			case 'blockquote_start':
				blockquote_start = true;

				if (options.allowBlockquote)
				{
					paragraph2 = paragraph;
					paragraph = [];
				}
				else
				{
					_skip = true;
				}

				//console.log(tok);
				break;
			case 'blockquote_end':

				if (options.allowBlockquote && blockquote_start && paragraph.length)
				{
					val = paragraph.join(eol);
					val = val.replace(/\s+$/g, '');

					if (!options.oldParseApi)
					{
						val = new RawObject(val, {
							type: 'blockquote',
							text: paragraph,

							paragraph: paragraph2,
						}) as any;
					}

					put(conf, keys, val, true, undefined, options);

					paragraph = [];
				}
				else
				{
					_skip = true;
				}

				blockquote_start = false;
				break;
			case 'paragraph':
				paragraph.push(tok.text);
				//console.log(tok);
				break;
			case 'code':
				val = val.replace(/\s+$/g, '');

				if (!options.oldParseApi)
				{
					val = new RawObject(val, tok) as any;
					(val as any as IRawObjectPlus).getRawData().paragraph = paragraph;
				}

				put(conf, keys, val, true, undefined, options);
				break;
			case 'table':
				put(conf, keys, null, null, { headers: tok.header, rows: tok.cells }, options);
				break;
			case 'html':
				val = val.replace(/\s+$/g, '');

				if (!options.oldParseApi)
				{
					val = new RawObject(val, tok) as any;
					(val as any as IRawObjectPlus).getRawData().paragraph = paragraph;
				}

				put(conf, keys, val, true, undefined, options);
				break;
			default:
				//console.log(tok);

				_skip = true;

				break;
		}

		if (!_skip && !['paragraph'].includes(tok.type))
		{
			paragraph = [];
		}

		last_tok = tok;
	});

	{
		let parent;
		let parent2 = conf;
		let parent3;

		for (let i in keys)
		{
			let k = keys[i];

			if (/^\d+$/.test(k))
			{
				// @ts-ignore
				let kk = keys[i-1];

				// @ts-ignore
				let parent = getobjectbyid(keys.slice(0, i-1), conf);
				// @ts-ignore
				let obj = getobjectbyid(keys.slice(0, i), conf);

				let ok = true;

				for (let j in obj as object)
				{
					if (!/^\d+$/.test(j))
					{
						ok = false;
						break;
					}
				}

				if (ok)
				{
					parent[kk] = Object.values(obj);
				}
			}

		}
	}

	return conf;
}

export function stringify(dataInput: unknown | IRawObjectPlus, level: number = 1, skip = [], k?): string
{
	let rs1: string[] = [];
	let rs2: string[] = [];

	let isRawObject: true | false;
	let data = dataInput;
	let desc;

	if (isRawObject = RawObject.isRawObject(dataInput))
	{
		let rawData = dataInput.getRawData();

		if (rawData.paragraph)
		{
			desc = rawData.paragraph.join(LF.repeat(2));
		}

		data = dataInput.getRawValue();

		isRawObject = true as true;
	}

	//console.log(k);

	if (Array.isArray(data))
	{
		if (k || k === 0)
		{
			rs2.push('#'.repeat(level) + '' + k + LF);

			data.forEach(function (value, index, array)
			{
				let bool = (!RawObject.isRawObject(value) && typeof value == 'object');

				rs2.push(stringify(value, level, [], bool ? index : null));
			})
		}
		else
		{
			data.forEach(function (value, index, array)
			{
				let bool = (!RawObject.isRawObject(value) && typeof value == 'object');

				rs1.push(stringify(value, level, [], bool ? index : null).replace(/\n+$/g, ''));
			});

			//rs1.push('');
		}
	}
	else if (typeof data == 'object')
	{
		if (k || k === 0)
		{
			rs1.push('#'.repeat(level) + ' ' + k + LF);
		}

		for (let k in data)
		{
			if (skip.includes(k))
			{
				continue;
			}

			let isRawObject = RawObject.isRawObject(data[k]);
			let row = isRawObject ? data[k].getRawValue() : data[k];

			if (Array.isArray(row))
			{
				rs2.push('#'.repeat(level) + ' ' + k + LF);
				rs2.push(stringify(row, level + 1));
			}
			else if (isPlainObject(row))
			{
				rs2.push('#'.repeat(level) + ' ' + k + LF);
				rs2.push(stringify(row, level + 1));
			}
			else if (isMoment(row))
			{
				rs1.push(`- ${k}: ${row.format()}`);
			}
			else if (isRawObject || typeof row == 'string' && /[\r\n]|^\s/g.test(row))
			{
				let lang: string;
				let val = row;

				val = val.replace(/^[\r\n]+|\s+$/g, '');

				if (isRawObject)
				{
					let rawData = data[k].getRawData() || {};

					if (rawData.type != 'html')
					{
						lang = rawData.lang;

						val = makeCodeBlock(val, lang);
					}
					else
					{
						val = LF + val + LF;
					}
				}
				else
				{
					val = makeCodeBlock(val, lang);
				}

				rs2.push('#'.repeat(level) + ' ' + k + LF);
				rs2.push(val);
			}
			else
			{
				rs1.push(`- ${k}: ${row}`);
			}
		}
	}
	else if (isRawObject || typeof data == 'string' && /[\r\n]|^\s/g.test(data))
	{
		if (k || k === 0)
		{
			rs2.push('#'.repeat(level) + ' ' + k + LF);
		}

		if (desc)
		{
			rs2.push(desc);
		}

		let val = data as string;

		val = val.replace(/^[\r\n]+|\s+$/g, '');

		if (isRawObject)
		{
			let rawData = (dataInput as IRawObjectPlus).getRawData() || {} as IRawObjectDataPlus;

			if (rawData.type != 'html')
			{
				val = makeCodeBlock(val, rawData.lang);
			}
			else
			{
				val = LF + val + LF;
			}
		}
		else
		{
			val = makeCodeBlock(val);
		}

		rs2.push(val);
	}
	else
	{
		if (desc)
		{
			rs1.push(desc);
		}

		rs1.push(`- ${ k || k === 0 ? k + ': ' : '' }${data}`);
	}

	let out = (rs1.concat([''].concat(rs2)).join(LF)).replace(/^\n+/g, '');

	if (level == 1)
	{
		out = out.replace(/^\n+|\s+$/g, '') + LF;
	}

	return out;
}

export default exports as typeof import('./core');

