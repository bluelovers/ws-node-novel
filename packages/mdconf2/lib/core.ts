/**
 * Created by user on 2020/1/15.
 */

import { Token, Tokens, InlineLexer, TokensList } from 'marked';
import { ITokenText2, RawObject, IRawObjectDataPlus } from './RawObject';
import { IOptionsParse, defaultOptionsParse } from '../core';

export function getobjectbyid<T extends unknown>(a: string[], conf: object): T
{
	let ret = conf as any;
	for (let i of a)
	{
		ret = ret[i];
	}
	return ret;
}

export interface ITable
{
	headers: Tokens.Table["header"],
	rows: Tokens.Table["cells"],
}

/**
 * Add `str` to `obj` with the given `keys`
 * which represents the traversal path.
 *
 * @api private
 */
export function put(obj,
	keys: string[],
	str: string,
	code?: boolean,
	table?: ITable,
	options: IOptionsParse = {},
	others: {
		type?: (Token | ITokenText2)["type"],
	} = {},
)
{
	let target = obj;
	let last;
	let key;

	for (let i = 0; i < keys.length; i++)
	{
		key = keys[i];
		last = target;
		target[key] = target[key] || {};
		target = target[key];
	}

	// code
	if (code)
	{
		if (!Array.isArray(last[key])) last[key] = [];
		last[key].push(str);
		return;
	}

	// table
	if (table)
	{
		if (!Array.isArray(last[key])) last[key] = [];
		for (let ri = 0; ri < table.rows.length; ri++)
		{
			let arrItem = {} as Record<string, string>;
			for (let hi = 0; hi < table.headers.length; hi++)
			{
				arrItem[normalize(table.headers[hi], options)] = table.rows[ri][hi];
			}
			last[key].push(arrItem);
		}
		return;
	}

	let isKey: boolean;
	let i: number = str.indexOf(':');

	if (options.filterObjectKey)
	{
		if (typeof options.filterObjectKey == 'function')
		{
			isKey = options.filterObjectKey(str, obj, others);
		}
		else
		{
			i = str.search(options.filterObjectKey);
			isKey = i != -1;
		}
	}

	// list
	if ((isKey === false || -1 == i || others.type == 'text2'))
	{
		if (!Array.isArray(last[key])) last[key] = [];
		last[key].push(str.trim());
		return;
	}

	// map
	key = normalize(str.slice(0, i), options);
	let val = str.slice(i + 1).trim();
	target[key] = val;
}

/**
 * Normalize `str`.
 */

export function normalize(str: string, options: IOptionsParse = {}): string
{
	let key = str.replace(/\s+/g, ' ');

	if (!options.disableKeyToLowerCase)
	{
		key = key.toLowerCase();
	}

	return key.trim();
}

export function makeCodeBlock(value, lang?: Tokens.Code["lang"])
{
	return `\n\`\`\`${lang || ''}\n${value}\n\`\`\`\n`;
}

export function createInlineLexer(toks: TokensList, options: IOptionsParse)
{
	let opts = Object.assign({}, defaultOptionsParse.markedOptions, options.markedOptions);

	// @ts-ignore
	let inline = new InlineLexer(toks.links, opts);

	return inline;
}
