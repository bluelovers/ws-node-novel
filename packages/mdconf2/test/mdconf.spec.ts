import * as testUtils from './test-utils';
import mdconf, { RawObject } from '../index';
import { parse } from '../core';
import { getTestCasesSync } from './test-utils';

describe(`mdconf integration tests`, () =>
{
	let allTestCases = getTestCasesSync();

	describe(`should pass integration tests`, () =>
	{
		allTestCases.forEach(testCase =>
		{
			describe('Test Case(s) for ' + testCase.testName, function ()
			{
				it('should match expected output with no options', () =>
				{
					let data = parse(testCase.md);

					expect(data).toMatchSnapshot();
					expect(mdconf.stringify(data)).toMatchSnapshot();

					let data3 = parse(mdconf.stringify(data));

					let data2 = RawObject.removeRawData(data);

					expect(data2).toStrictEqual(testCase.json);
					expect(data2).toStrictEqual(RawObject.removeRawData(data3));

					expect(data2).toMatchSnapshot();
					expect(data3).toMatchSnapshot();

				});
			});
		});

	});

})
