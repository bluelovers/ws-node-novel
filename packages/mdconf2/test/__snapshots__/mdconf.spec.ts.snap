// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mdconf integration tests should pass integration tests Test Case(s) for blocks should match expected output with no options 1`] = `
Object {
  "styles": Array [
    "body {
  padding: 50px;
}",
    "button {
  padding: 5px 10px;
}",
  ],
}
`;

exports[`mdconf integration tests should pass integration tests Test Case(s) for blocks should match expected output with no options 2`] = `
"# styles

  Just some example styles:

\`\`\`
body {
  padding: 50px;
}
\`\`\`
 And some styles for buttons:

\`\`\`
button {
  padding: 5px 10px;
}
\`\`\`
"
`;

exports[`mdconf integration tests should pass integration tests Test Case(s) for blocks should match expected output with no options 3`] = `
Object {
  "styles": Array [
    "body {
  padding: 50px;
}",
    "button {
  padding: 5px 10px;
}",
  ],
}
`;

exports[`mdconf integration tests should pass integration tests Test Case(s) for blocks should match expected output with no options 4`] = `
Object {
  "styles": Array [
    "body {
  padding: 50px;
}",
    "button {
  padding: 5px 10px;
}",
  ],
}
`;

exports[`mdconf integration tests should pass integration tests Test Case(s) for code should match expected output with no options 1`] = `
Object {
  "styles": Array [
    "body {
  padding: 50px;
}",
    "button {
  padding: 5px 10px;
}",
  ],
}
`;

exports[`mdconf integration tests should pass integration tests Test Case(s) for code should match expected output with no options 2`] = `
"# styles

  Just some example styles:

\`\`\`scss
body {
  padding: 50px;
}
\`\`\`
 And some styles for buttons:

\`\`\`css
button {
  padding: 5px 10px;
}
\`\`\`
"
`;

exports[`mdconf integration tests should pass integration tests Test Case(s) for code should match expected output with no options 3`] = `
Object {
  "styles": Array [
    "body {
  padding: 50px;
}",
    "button {
  padding: 5px 10px;
}",
  ],
}
`;

exports[`mdconf integration tests should pass integration tests Test Case(s) for code should match expected output with no options 4`] = `
Object {
  "styles": Array [
    "body {
  padding: 50px;
}",
    "button {
  padding: 5px 10px;
}",
  ],
}
`;

exports[`mdconf integration tests should pass integration tests Test Case(s) for complex should match expected output with no options 1`] = `
Object {
  "defaults": Object {
    "s3": Object {
      "api key": "111111",
      "buckets": Object {
        "assets": "myapp-assets",
        "avatars": "myapp-avatars",
        "files": "myapp-files",
      },
      "secret": "222222",
    },
    "upload": Object {
      "dir": "/tmp",
      "max": "200mb",
      "thumbnail sizes": Array [
        "50x50",
        "300x300",
        "600x600",
        "900x900",
      ],
    },
  },
  "production": Object {
    "sites": Array [
      Object {
        "build": "passing",
        "coverage": "94%",
        "hostname": "google.com",
      },
      Object {
        "build": "passing",
        "coverage": "97%",
        "hostname": "facebook.com",
      },
      Object {
        "build": "failed",
        "coverage": "81%",
        "hostname": "twitter.com",
      },
      Object {
        "build": "unkown",
        "coverage": "0%",
        "hostname": "myspace.com",
      },
    ],
    "upload": Object {
      "max": "200mb",
    },
  },
}
`;

exports[`mdconf integration tests should pass integration tests Test Case(s) for complex should match expected output with no options 2`] = `
"# defaults

## upload

- max: 200mb
- dir: /tmp

### thumbnail sizes

- 50x50
- 300x300
- 600x600
- 900x900

## s3

- api key: 111111
- secret: 222222

### buckets

- avatars: myapp-avatars
- assets: myapp-assets
- files: myapp-files

# production

## upload

- max: 200mb

## sites

### 0

- hostname: google.com
- build: passing
- coverage: 94%
### 1

- hostname: facebook.com
- build: passing
- coverage: 97%
### 2

- hostname: twitter.com
- build: failed
- coverage: 81%
### 3

- hostname: myspace.com
- build: unkown
- coverage: 0%
"
`;

exports[`mdconf integration tests should pass integration tests Test Case(s) for complex should match expected output with no options 3`] = `
Object {
  "defaults": Object {
    "s3": Object {
      "api key": "111111",
      "buckets": Object {
        "assets": "myapp-assets",
        "avatars": "myapp-avatars",
        "files": "myapp-files",
      },
      "secret": "222222",
    },
    "upload": Object {
      "dir": "/tmp",
      "max": "200mb",
      "thumbnail sizes": Array [
        "50x50",
        "300x300",
        "600x600",
        "900x900",
      ],
    },
  },
  "production": Object {
    "sites": Array [
      Object {
        "build": "passing",
        "coverage": "94%",
        "hostname": "google.com",
      },
      Object {
        "build": "passing",
        "coverage": "97%",
        "hostname": "facebook.com",
      },
      Object {
        "build": "failed",
        "coverage": "81%",
        "hostname": "twitter.com",
      },
      Object {
        "build": "unkown",
        "coverage": "0%",
        "hostname": "myspace.com",
      },
    ],
    "upload": Object {
      "max": "200mb",
    },
  },
}
`;

exports[`mdconf integration tests should pass integration tests Test Case(s) for complex should match expected output with no options 4`] = `
Object {
  "defaults": Object {
    "s3": Object {
      "api key": "111111",
      "buckets": Object {
        "assets": "myapp-assets",
        "avatars": "myapp-avatars",
        "files": "myapp-files",
      },
      "secret": "222222",
    },
    "upload": Object {
      "dir": "/tmp",
      "max": "200mb",
      "thumbnail sizes": Array [
        "50x50",
        "300x300",
        "600x600",
        "900x900",
      ],
    },
  },
  "production": Object {
    "sites": Array [
      Object {
        "build": "passing",
        "coverage": "94%",
        "hostname": "google.com",
      },
      Object {
        "build": "passing",
        "coverage": "97%",
        "hostname": "facebook.com",
      },
      Object {
        "build": "failed",
        "coverage": "81%",
        "hostname": "twitter.com",
      },
      Object {
        "build": "unkown",
        "coverage": "0%",
        "hostname": "myspace.com",
      },
    ],
    "upload": Object {
      "max": "200mb",
    },
  },
}
`;

exports[`mdconf integration tests should pass integration tests Test Case(s) for empty should match expected output with no options 1`] = `Object {}`;

exports[`mdconf integration tests should pass integration tests Test Case(s) for empty should match expected output with no options 2`] = `
"
"
`;

exports[`mdconf integration tests should pass integration tests Test Case(s) for empty should match expected output with no options 3`] = `Object {}`;

exports[`mdconf integration tests should pass integration tests Test Case(s) for empty should match expected output with no options 4`] = `Object {}`;

exports[`mdconf integration tests should pass integration tests Test Case(s) for html should match expected output with no options 1`] = `
Object {
  "html": Array [
    "<!-- js
111111111
<br/>
-->",
    "2222",
  ],
}
`;

exports[`mdconf integration tests should pass integration tests Test Case(s) for html should match expected output with no options 2`] = `
"# html

<!-- js
111111111
<br/>
-->
\`\`\`
2222
\`\`\`
"
`;

exports[`mdconf integration tests should pass integration tests Test Case(s) for html should match expected output with no options 3`] = `
Object {
  "html": Array [
    "<!-- js
111111111
<br/>
-->",
    "2222",
  ],
}
`;

exports[`mdconf integration tests should pass integration tests Test Case(s) for html should match expected output with no options 4`] = `
Object {
  "html": Array [
    "<!-- js
111111111
<br/>
-->",
    "2222",
  ],
}
`;

exports[`mdconf integration tests should pass integration tests Test Case(s) for html2 should match expected output with no options 1`] = `
Object {
  "html": Array [
    "777",
    "789",
    "3333




444

444",
    "<!-- js
111111111
<br/>
-->",
    "111",
    "2222",
  ],
}
`;

exports[`mdconf integration tests should pass integration tests Test Case(s) for html2 should match expected output with no options 2`] = `
"# html

555555555555555

\`\`\`
777
\`\`\`
\`\`\`
789
\`\`\`
\`\`\`
3333




444

444
\`\`\`
555

66

777

888

999

444

<!-- js
111111111
<br/>
-->
\`\`\`
111
\`\`\`
\`\`\`
2222
\`\`\`
"
`;

exports[`mdconf integration tests should pass integration tests Test Case(s) for html2 should match expected output with no options 3`] = `
Object {
  "html": Array [
    "777",
    "789",
    "3333




444

444",
    "<!-- js
111111111
<br/>
-->",
    "111",
    "2222",
  ],
}
`;

exports[`mdconf integration tests should pass integration tests Test Case(s) for html2 should match expected output with no options 4`] = `
Object {
  "html": Array [
    "777",
    "789",
    "3333




444

444",
    "<!-- js
111111111
<br/>
-->",
    "111",
    "2222",
  ],
}
`;

exports[`mdconf integration tests should pass integration tests Test Case(s) for link should match expected output with no options 1`] = `
Object {
  "link": Array [
    "[dip.jp](https://narou18.dip.jp/search.php?text=n0153ce&novel=all&genre=all&new_genre=all&length=0&down=0&up=100) - 小説家になろう　更新情報検索",
    "[我的同学是姬骑士吧](https://tieba.baidu.com/f?kw=%E6%88%91%E7%9A%84%E5%90%8C%E5%AD%A6%E6%98%AF%E5%A7%AC%E9%AA%91%E5%A3%AB&ie=utf-8 \\"我的同学是姬骑士\\")",
    "[姫騎士がクラスメート！　THE COMIC](http://www.comic-valkyrie.com/modules/web_valkyrie/classmate/)",
    "https://www.lightnovel.cn/forum.php?mod=viewthread&tid=922044",
  ],
}
`;

exports[`mdconf integration tests should pass integration tests Test Case(s) for link should match expected output with no options 2`] = `
"# link

- [dip.jp](https://narou18.dip.jp/search.php?text=n0153ce&novel=all&genre=all&new_genre=all&length=0&down=0&up=100) - 小説家になろう　更新情報検索
- [我的同学是姬骑士吧](https://tieba.baidu.com/f?kw=%E6%88%91%E7%9A%84%E5%90%8C%E5%AD%A6%E6%98%AF%E5%A7%AC%E9%AA%91%E5%A3%AB&ie=utf-8 \\"我的同学是姬骑士\\")
- [姫騎士がクラスメート！　THE COMIC](http://www.comic-valkyrie.com/modules/web_valkyrie/classmate/)
- https://www.lightnovel.cn/forum.php?mod=viewthread&tid=922044
"
`;

exports[`mdconf integration tests should pass integration tests Test Case(s) for link should match expected output with no options 3`] = `
Object {
  "link": Array [
    "[dip.jp](https://narou18.dip.jp/search.php?text=n0153ce&novel=all&genre=all&new_genre=all&length=0&down=0&up=100) - 小説家になろう　更新情報検索",
    "[我的同学是姬骑士吧](https://tieba.baidu.com/f?kw=%E6%88%91%E7%9A%84%E5%90%8C%E5%AD%A6%E6%98%AF%E5%A7%AC%E9%AA%91%E5%A3%AB&ie=utf-8 \\"我的同学是姬骑士\\")",
    "[姫騎士がクラスメート！　THE COMIC](http://www.comic-valkyrie.com/modules/web_valkyrie/classmate/)",
    "https://www.lightnovel.cn/forum.php?mod=viewthread&tid=922044",
  ],
}
`;

exports[`mdconf integration tests should pass integration tests Test Case(s) for link should match expected output with no options 4`] = `
Object {
  "link": Array [
    "[dip.jp](https://narou18.dip.jp/search.php?text=n0153ce&novel=all&genre=all&new_genre=all&length=0&down=0&up=100) - 小説家になろう　更新情報検索",
    "[我的同学是姬骑士吧](https://tieba.baidu.com/f?kw=%E6%88%91%E7%9A%84%E5%90%8C%E5%AD%A6%E6%98%AF%E5%A7%AC%E9%AA%91%E5%A3%AB&ie=utf-8 \\"我的同学是姬骑士\\")",
    "[姫騎士がクラスメート！　THE COMIC](http://www.comic-valkyrie.com/modules/web_valkyrie/classmate/)",
    "https://www.lightnovel.cn/forum.php?mod=viewthread&tid=922044",
  ],
}
`;

exports[`mdconf integration tests should pass integration tests Test Case(s) for simple should match expected output with no options 1`] = `
Object {
  "uploads": Object {
    "dir": "/tmp/uploads",
    "max": "200mb",
    "min": "1kb",
  },
}
`;

exports[`mdconf integration tests should pass integration tests Test Case(s) for simple should match expected output with no options 2`] = `
"# uploads

- max: 200mb
- min: 1kb
- dir: /tmp/uploads
"
`;

exports[`mdconf integration tests should pass integration tests Test Case(s) for simple should match expected output with no options 3`] = `
Object {
  "uploads": Object {
    "dir": "/tmp/uploads",
    "max": "200mb",
    "min": "1kb",
  },
}
`;

exports[`mdconf integration tests should pass integration tests Test Case(s) for simple should match expected output with no options 4`] = `
Object {
  "uploads": Object {
    "dir": "/tmp/uploads",
    "max": "200mb",
    "min": "1kb",
  },
}
`;

exports[`mdconf integration tests should pass integration tests Test Case(s) for table should match expected output with no options 1`] = `
Object {
  "sites": Array [
    Object {
      "build": "passing",
      "coverage": "94%",
      "host": "google.com",
    },
    Object {
      "build": "passing",
      "coverage": "97%",
      "host": "facebook.com",
    },
    Object {
      "build": "failed",
      "coverage": "81%",
      "host": "twitter.com",
    },
    Object {
      "build": "unkown",
      "coverage": "0%",
      "host": "myspace.com",
    },
  ],
}
`;

exports[`mdconf integration tests should pass integration tests Test Case(s) for table should match expected output with no options 2`] = `
"# sites

## 0

- host: google.com
- build: passing
- coverage: 94%
## 1

- host: facebook.com
- build: passing
- coverage: 97%
## 2

- host: twitter.com
- build: failed
- coverage: 81%
## 3

- host: myspace.com
- build: unkown
- coverage: 0%
"
`;

exports[`mdconf integration tests should pass integration tests Test Case(s) for table should match expected output with no options 3`] = `
Object {
  "sites": Array [
    Object {
      "build": "passing",
      "coverage": "94%",
      "host": "google.com",
    },
    Object {
      "build": "passing",
      "coverage": "97%",
      "host": "facebook.com",
    },
    Object {
      "build": "failed",
      "coverage": "81%",
      "host": "twitter.com",
    },
    Object {
      "build": "unkown",
      "coverage": "0%",
      "host": "myspace.com",
    },
  ],
}
`;

exports[`mdconf integration tests should pass integration tests Test Case(s) for table should match expected output with no options 4`] = `
Object {
  "sites": Array [
    Object {
      "build": "passing",
      "coverage": "94%",
      "host": "google.com",
    },
    Object {
      "build": "passing",
      "coverage": "97%",
      "host": "facebook.com",
    },
    Object {
      "build": "failed",
      "coverage": "81%",
      "host": "twitter.com",
    },
    Object {
      "build": "unkown",
      "coverage": "0%",
      "host": "myspace.com",
    },
  ],
}
`;
